framework/src/aifw/AIModelService.cpp

```diff
@@ -0,0 +1,201 @@
+/****************************************************************************
+ *
+ * Copyright 2023 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include "aifw/timer.h"
+#include <stdlib.h>
+#include <memory>
+#include <pthread.h>
+#include "aifw/aifw.h"
+#include "aifw/aifw_log.h"
+#include "aifw/AIModelService.h"
+#include "aifw/AIInferenceHandler.h"
+
+namespace aifw {
+
+AIModelService::AIModelService(CollectRawDataListener collectRawDataCallback, std::shared_ptr<AIInferenceHandler> inferenceHandler) :
+	mInterval(0), mServiceRunning(false), mInferenceHandler(inferenceHandler), mCollectRawDataCallback(collectRawDataCallback), mTimer(NULL)
+{
+}
+
+AIModelService::~AIModelService()
+{
+	freeTimer();
+	AIFW_LOGV("model service object destoyed");
+}
+
+static void *destroyTimer(void *arg)
```

Comment:
timer_stop actually stops timer and exits thread, so shouldn't this thread be used for timer_stop